go.mod

```diff
@@ -8,18 +8,16 @@ replace (
 	github.com/apparentlymart/go-textseg/v12 => github.com/apparentlymart/go-textseg/v12 v12.0.0
 	github.com/godbus/dbus => github.com/godbus/dbus v5.0.1+incompatible
 	github.com/golang/protobuf => github.com/golang/protobuf v1.3.4
+	github.com/hashicorp/go-discover => github.com/hashicorp/go-discover v0.0.0-20200812215701-c4b85f6ed31f
```

Comment:
My go-mod knowledge is weak - This seems like the same version as below; what motivates adding an explicit replace statement here.

Reply:
This would prevent the `github.com/hashicorp/go-discover` module from being updated in the future, unless explicitly done in (or removed from) the `replace` block. I suspect that isn't the intent and this could be removed.

Reply:
It replaces **all** instances of go-discover with that one. Otherwise, the old version is picked up, notably by Consul, and pulls back gocheck from launchpad.

Reply:
I see - so without replace, we may have multiple go-discover instances, one that's used by nomad, but another version may be used by nomad's dependencies.

Reply:
When you look at the `go.sum`, the oldest one is used. AFAIK, it doesn't mix multiple versions.

> "If an old package and a new package have the same import path,
the new package must be backwards compatible with the old package."
> 
> Because the go command assumes the import compatibility rule,
a module definition can only set the minimum required version of one
of its dependencies: it cannot set a maximum or exclude selected versions.
Still, the import compatibility rule is not a guarantee: it may be that
v1.5.4 is buggy and not a backwards-compatible replacement for v1.5.3.
Because of this, the go command never updates from an older version
to a newer version of a module unasked.
>
> https://github.com/golang/go/blob/dbde566219336e84360b4a38da10b5f63b19021e/src/cmd/go/internal/modload/help.go#L274